### PR TITLE
add support for next question link with LinkProvider method (NumericA…

### PR DESCRIPTION
**Pull request by mnembrini**
…nswer example)

### CODE REVIEW

File: rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/ContractReader.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  private static final Logger log = LoggerFactory.getLogger(ContractReader.class);
+
+  private Map<Class, ResourceContract> contractStore = new HashMap<>();
+  private final ObjectMapper objectMapper;
+
+  public ContractReader(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  public Optional<ResourceContract> fetchContract(Object resourceBean) {
```
Comment (saw303) [inline rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/ContractReader.java]: Maybe a candidate for `@Cachable`?

File: rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/HateoasResponseFilter.java [hunk 2]
```diff
@@ -71,6 +72,11 @@
 @Filter("${restapi.hateoas.filter.uri}")
 @Requires(property = "restapi.hateoas.filter.enabled")
 public class HateoasResponseFilter implements HttpServerFilter {
+  private final List<LinkProvider> linkProviderList;
+
+  public HateoasResponseFilter(List<LinkProvider> linkProviderList) {
```
Comment (saw303) [inline rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/HateoasResponseFilter.java]: What happens if a project does not provide any `LinkProvider`? The startup of that application should not fail. Meaning the `linkProviderList` should be consider as an optional dependency of `HateoasResponseFilter`

File: rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/LinkProvider.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package ch.silviowangler.rest.micronaut;
+
+import ch.silviowangler.rest.model.ResourceLink;
+import io.micronaut.web.router.UriRouteMatch;
+import java.util.List;
+
+public interface LinkProvider {
```
Comment (saw303) [inline rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/LinkProvider.java]: Few javadoc comments about this interface

File: rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/ContractReader.java (hunk 1 above)
Comment (mnembrini) [reply rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/ContractReader.java]: 👍

File: rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/HateoasResponseFilter.java (hunk 2 above)
Comment (mnembrini) [reply rest-api-micronaut/src/main/java/ch/silviowangler/rest/micronaut/HateoasResponseFilter.java]: Micronaut should inject an empty list if there are no beans of type LinkProvider

**Pull request merged**